### DISCUSSION TITLE
Markdown mermaid support

### DISCUSSION THREAD

**Discussion by Stratis-Dermanoutsos** [withastro/roadmap#267] [Proposal]
So, I am using the documentation template for a personal documentation page.

I noticed that the markdown pages do not support **mermaid** code blocks.

Specifically, the CMD logs:

```plaintext
The language "mermaid" doesn't exist, falling back to plaintext.
```

Example:

```plaintext
```mermaid
 graph TD;
   a-->b;
   b-->c;
   c-->d;
\```
```

> The \` is escaped here for GitHub to correctly format the text. I did not escape the code block in the actual code.

I know this is a very minor feature but it'd be very helpful to have.

**Comment by stereobooster** [comment]
Do you expect code highlighting or [it being transformed to a diagram](https://github.com/withastro/starlight/discussions/1259)?

**Comment by chrisirhc** [comment]
https://github.com/withastro/roadmap/discussions/1124 might be related if this has to do with rendering.
